### DISCUSSION TITLE
How to make the same MITM request with HTTP Client C#

### DISCUSSION THREAD

**Discussion by teamothyz** [mitmproxy/mitmproxy#6105] [Q&A]
Hi Guys, I'm using MITM to find API for automation. But I have a problem with http client request (c#). I can't make the request to destination normally or using 3rd proxy (http status 403), but when using MITM proxy (turn on it on Windows) and make a request, it was done with http status 200. I don't know why... can anyone help me to explain what differences here? And how can I make the same MITM request to receive the response?

Thanks for reading!

**Comment by sujaldev** [comment]
If you're getting an HTTP 403, you're most likely not sending some header responsible for auth.

**Reply by teamothyz** [reply]
I've tried to add all headers value same as the original request. With the same headers, I got success status when redirect to MITM but got access denied (403) status when not redirect to MITM. 
P/s: My destination using Akamai to prevent bot request.

**Reply by teamothyz** [reply]
So could you help me figure out what I did wrong?

**Reply by sujaldev** [reply]
> With the same headers, I got success status when redirect to MITM but got access denied (403) status when not redirect to MITM.

What do you mean? Are you saying you get an HTTP 403 when **not** using mitmproxy?
